### ISSUE TITLE
[java] SingularField: False-positive when preceded by synchronized block

### ISSUE THREAD

**Issue by rdicroce** [pmd/pmd#5162]
**Affects PMD Version:** 

7.4.0

**Rule:**

[SingularField](https://docs.pmd-code.org/latest/pmd_rules_java_design.html#singularfield)

**Description:**

SingularField incorrectly reports that a field could be a local variable if all methods where the variable is used contain a synchronized block _before_ the use of the field.

The problem does not occur if the synchronized block comes _after_ the use of the field, or if the synchronized block is removed. The problem also does not occur if there's another method in the class that uses the field and does not have a synchronized block before the use of the field.

**Code Sample demonstrating the issue:**

```java
package test;

import java.util.List;

public class PmdBug {
	
	private List<String> sessions;

	public PmdBug(List<String> sessions) {
		this.sessions = sessions;
	}
	
	public void doStuff() {
		synchronized (this) {
			
		}
		
		sessions.stream().forEach(System.out::println);
	}
	
}
```

**Expected outcome:**

PMD should not report any violations. But it reports a violation at line 7.

**Running PMD through:**

Eclipse and Maven

**Comment by rdicroce** [comment]
Thanks for the fix!
